### PR TITLE
[SU-XX] updates icons, colors and providers

### PR DESCRIPTION
**Pull request by Zyruks**
## Changes Made 🎉

- [x] refactor: improved code readability and organization
- [x] chore: updated dependencies and configuration files

### Describe Changes
Update Button colors
Add new Icon Squares2x2
generalize `ToggleGroup` with TypeScript generics
Replace specific type annotations with TypeScript generics for greater flexibility.
Change function components to use function declaration syntax for consistency.
Update type and interface definitions to support generics.

### Button and Square2x2
image

Before 
image

Update provider to generics to accept enums
image

## Checklist ✅

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/contexts/ToggleGroupContext.tsx [hunk 1]
```diff
@@ -1,38 +1,32 @@
...
 /*  CONTEXT DEFINITION  */
-export const ToggleGroupContext = createContext<ToggleGroupContextType>({
+export const ToggleGroupContext = createContext<ToggleGroupContextType<any>>({
   value: null,
   onChange: () => {},
 });
 
 /* PROVIDER DEFINITION */
-export const ToggleGroupProvider = ({
+export function ToggleGroupProvider<T>({
```
Comment (serudda) [inline src/contexts/ToggleGroupContext.tsx:24]: Por que lo cambias a function?

File: src/contexts/ToggleGroupContext.tsx [hunk 2]
```diff
@@ -46,11 +40,11 @@ export const ToggleGroupProvider = ({
       <div {...props}>{children}</div>
     </ToggleGroupContext.Provider>
   );
-};
+}
 
 /*   EXPORT USE METHOD   */
-export const useToggleGroup = (): ToggleGroupContextType => {
+export function useToggleGroup<T>(): ToggleGroupContextType<T> {
```
Comment (serudda) [inline src/contexts/ToggleGroupContext.tsx:46]: Por que el cambio a function?

**Review by serudda — CHANGES_REQUESTED** [review]

File: src/contexts/ToggleGroupContext.tsx (hunk 1 above)
Comment (Zyruks) [reply src/contexts/ToggleGroupContext.tsx]: Done!

File: src/contexts/ToggleGroupContext.tsx (hunk 2 above)
Comment (Zyruks) [reply src/contexts/ToggleGroupContext.tsx]: Done!

**Pull request merged**